### PR TITLE
SL1K4, SL2K4 and ST4K4

### PR DESCRIPTION
**Pull request by ghalym**
DEV: Included the implementation, NC axes and etherCat devices for SL2K4 and ST4K4.

### CODE REVIEW

File: plc-tmo-motion/tmo_motion/POUs/PRG_SL1K4_SCATTER.TcPOU [hunk 1]
```diff
@@ -4,7 +4,7 @@
     <Declaration><![CDATA[PROGRAM PRG_SL1K4_SCATTER
 VAR
     {attribute 'pytmc' := '
-        pv: SL1K4:SCATTER
+        pv: SL2K4:SCATTER
```
Comment (ZLLentz) [inline plc-tmo-motion/tmo_motion/POUs/PRG_SL1K4_SCATTER.TcPOU:7]: I think this is a typo, this is the SL2K4 program

**Comment by ZLLentz** [comment]
The two merge conflicts are because this branch is missing the most recent PR's commits

File: plc-tmo-motion/tmo_motion/POUs/PRG_SL1K4_SCATTER.TcPOU (hunk 1 above)
Comment (ghalym) [reply plc-tmo-motion/tmo_motion/POUs/PRG_SL1K4_SCATTER.TcPOU]: fixed in the new pull request

**Comment by ZLLentz** [comment]
Is this branch running on the TMO PLC right now? If so, it is missing some MPS fixes I put in for the WFS

**Comment by ghalym** [comment]
No, I didn't push these changes on the PLC. I wasn't sure what is the status of the code on the PLC.
I  hope we can push it next PAMM. 
I am working on a homing procedure for all the blades.

**Comment by ZLLentz** [comment]
Ok, sounds good. This has a merge conflict in the plcproj and tmc which I am happy to help resolve.

**Comment by ZLLentz** [comment]
Merge conflict looks like it has gone away and the diffs look clean

**Comment by ZLLentz** [comment]
Actually, according to the travis, the tmc is malformed. But we can rebuild this along with the IOC.

**Review by ZLLentz — APPROVED** [review]

**Pull request merged**